<launch>
  <arg name="id"          default="right" />
  <arg name="leg_model"   default="$(find leg_as_robot)/urdf/$(arg id)/leg.create.xacro"/>
  <arg name="virtual_leg" default="true" />
	<arg name="freq"        default="500.0"/>  
  <arg name="rviz"        default="true"/>
  <arg name="rvizconfig"  default="$(find leg_as_robot)/rviz/$(arg id)/leg_alone.rviz" />

  <arg name="offset_x"    default="0.0"/>  
  <arg name="offset_y"    default="0.0"/>
  <arg name="offset_z"    default="0.0"/>

  <group ns="$(arg id)_leg">
        
      <param name="robot_description" command="$(find xacro)/xacro --inorder $(arg leg_model)"/>

      <group if="$(arg virtual_leg)">
        <node name="virtual_leg_joint_publisher" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" output="screen">
        <param name="publish_frequency" value="$(arg freq)"/>
        <remap from="joint_states" to="leg_joint_publisher/joint_states" />
        </node>
      </group>	  

      <group unless="$(arg virtual_leg)">
        <node name="leg_joint_publisher" pkg="leg_as_robot" type="legRobot" output="screen">
          <param name="legID" value="$(arg id)"/>
          <param name="publish_frequency" value="$(arg freq)"/>
          <remap from="joint_states" to="leg_joint_publisher/joint_states" />
        </node>
      </group>

      <node name="leg_state_publisher" pkg="robot_state_publisher" type="state_publisher">
        <param name="publish_frequency" value="$(eval 2*arg('freq'))"/>
        <remap from="joint_states" to="leg_joint_publisher/joint_states" />
      </node>

  </group>

  <group if="$(arg rviz)"> 
      <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)"/> 
  </group>

  <node pkg="tf2_ros" type="static_transform_publisher" name="chair_to_$(arg id)_leg" args="$(arg offset_x)  $(arg offset_y) $(arg offset_z) 0 0 0 1 /chair_base_link $(arg id)_leg_hip_base_link" />  

</launch>
